common/bit_array_test.go

```diff
@@ -164,3 +164,26 @@ func TestEmptyFull(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdateNeverPanics(t *testing.T) {
+	newRandBitArray := func(n int) *BitArray {
+		ba, _ := randBitArray(n)
+		return ba
+	}
+	pairs := []struct {
+		a, b *BitArray
+	}{
+		{nil, nil},
+		{newRandBitArray(10), newRandBitArray(12)},
+		{newRandBitArray(0), NewBitArray(10)},
+		{nil, NewBitArray(10)},
+		{nil, newRandBitArray(64)},
+		{newRandBitArray(63), newRandBitArray(64)},
+	}
```

Comment:
Could you simplify the test cases:
* nil, nil,
* newRandBitArray, newRandBitArray
* newRandBitArray, nil
* nil, newRandBitArray

That way we test all the possible combination while also making it easier for developers to read and understand this test.

Reply:
Great idea, thanks @adrianbrink, I'll update that right now.

Reply:
Done, with https://github.com/tendermint/tmlibs/pull/136/commits/85be26c675b05a2a75c856f7c22b446d8df1c944